dindex_store/discovery_index.py

```diff
@@ -0,0 +1,178 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Dict, List
+
+
+class EdgeType(Enum):
+    ATTRIBUTE_SYNTACTIC_SIMILARITY = 0
+    COLUMN_CONTENT_JACCARD_SIMILARITY = 1
+    SYNTACTIC_INCLUSION_DEPENDENCY = 2
+
+
+class DiscoveryIndex:
+    """
+    DiscoveryIndex stores profiles generated by the profiler, a graph that
+    describes relationships between profiles, and a full text search index on
+    the original input data.
+
+    :param profile_index: ProfileIndex
+    :param graph_idnex: GraphIndex
+    :param fts_index: FullTextSearchIndex
+    """
+
+    def __init__(self, profile_index, graph_index, fts_index=None) -> None:
+        self.__profile_index = profile_index
+        self.__graph_index = graph_index
+        self.__fts_index = fts_index
+
+    # ----------------------------------------------------------------------
+    # Modify Methods
+
+    def add_node(self, node: Dict) -> bool:
```

Comment:
Should this be called add_profile instead? (a profile is a node in the graph index, but it still represents a profile, and that's also what it is in the profile index)